route/route-free-trial-request.js

```diff
@@ -1,13 +1,119 @@
 const bodyParse = require('body-parser').json();
 const Student = require('./../schemas/student');
+const Teacher = require('./../schemas/teacher');
+const makeAvailability = require('./../lib/makeAvailabilityObject');
+const availableTeachers = require('./../lib/filterTeachersByAvailability');
+const createAddress = require('./../lib/createAddrString');
+const findDistance = require('./../lib/distanceMatrixAPI');
+const createTimeSlot = require('./../lib/createTimeSlot');
 
 module.exports = router => {
   router.post('*api/free-trial-request', bodyParse, (req, res) => {
+    const postResponse = {};
     const bodyParams = Object.keys(req.body);
-    // Need to validate that bodyParams contains all required params
+
     if (!bodyParams.length) {
-      return res.status(400).send('Bad Request body');
+      return res.status(400).send('Bad Request: Request must include body');
+    }
+
+    const {
+      addressOne,
+      city,
+      state,
+      zip,
+      instrument,
+      musicStyle,
+      allergies,
+      specialNeeds,
+      comments,
+      experienceLevel,
+      availability,
+    } = req.body;
+
+    if (
```

Comment:
This method of request validation may cause some problems. For example, if they don't have any comments and `req.body.comments` comes across as null or an empty string, they would get an error response. But we can come back to this later - this can work for now.

Reply:
I agree here.  I removed allergies, specialNeeds, and comments as required fields, since those are free form text that the student can omit.  I also removed the 'required' flag on the student schema for these.